configs/training.py

```diff
@@ -30,6 +30,8 @@ class train_config:
     quantization: bool = False
     one_gpu: bool = False
     save_model: bool = True
+    use_cosine_scheduler: bool = False
```

Comment:
There are many different schedulers, can we make this an easily extensible enum or similar to avoid having n use_X_scheduler bools? Given that all need different parameters a separate config dataclass for schedulers would work as well and fit our pattern. Compare datasets